File: algorithm/2022 financial/2.js
function solution(matches, me, opp) {
  let answer = 0;

  let max = Math.max(...matches);
  let left_child = [];
  let right_child = [];

  for (let i = 0; i <= max; i++) {
    left_child[i] = 0;
    right_child[i] = 0;
  }

  matches.forEach((el, idx) => {
    if (left_child[el] === 0) left_child[el] = idx;
    else right_child[el] = idx;
  });

  while (me !== opp) {
    me = matches[me];
    opp = matches[opp];

    if (left_child[me] !== 0 && right_child[me] !== 0) answer += 1;
  }

  return answer;
}

File: algorithm/2022 financial/2_.py
def solution(match, myLoc, oppLoc):
    ans = 0

    l = [0] * (max(match) + 1)
    r = [0] * (max(match) + 1)

    for idx, i in enumerate(match):
        if l[i] == 0:
            l[i] = idx
        else:
            r[i] = idx

    print(l)
    print(r)

    while(myLoc != oppLoc):
        myLoc = match[myLoc]
        oppLoc = match[oppLoc]

        if l[myLoc] != 0 and r[myLoc] != 0:
            ans += 1
    print(ans)


matches = [-1, 0, 0, 1, 2, 2, 3, 3, 4,4, 5, 5,6, 7, 8, 8, 9, 10, 10, 11, 11]
me = 16
opp = 13